src/Repeater/DefaultRepeater.ts

```
@@ -0,0 +1,147 @@
+import { logger } from '../Utils';
+import {
+  Repeater,
+  RepeaterEventHandler,
+  RepeaterDeployedEvent,
+  RepeaterEvents
+} from './Repeater';
+import { inject, injectable } from 'tsyringe';
+import io, { Socket } from 'socket.io-client';
+import parser from 'socket.io-msgpack-parser';
+import { hostname } from 'os';
+import { once } from 'events';
+
+export interface DefaultRepeaterOptions {
+  readonly uri: string;
+  readonly token: string;
+}
+
+export const DefaultRepeaterOptions: unique symbol = Symbol(
+  'DefaultRepeaterOptions'
+);
+
+@injectable()
+export class DefaultRepeater implements Repeater {
+  private latestReconnectionError?: Error;
+  private readonly DEFAULT_RECONNECT_TIMES = 3;
+  private socket?: Socket;
+
+  constructor(
+    @inject(DefaultRepeaterOptions)
+    private readonly options: DefaultRepeaterOptions
+  ) {}
+
+  public disconnect() {
+    if (!this.socket) {
+      throw new Error('Not connected');
+    }
```

You should silently return the control If the socket is not connected